packages/flutter/lib/src/material/combo_box.dart

```diff
@@ -0,0 +1,747 @@
+    double width = 0;
+    while (child != null) {
+      if (child == firstChild) {
+        final _ComboBoxBodyParentData childParentData = child.parentData! as _ComboBoxBodyParentData;
+        child = childParentData.nextSibling;
+        continue;
+      }
+      final double maxIntrinsicWidth = child.getMinIntrinsicWidth(height);
+      if (child == lastChild) {
+        width += maxIntrinsicWidth;
+      }
+      if (child == childBefore(lastChild!)) {
+        width += maxIntrinsicWidth;
+      }
+      width = math.max(width, maxIntrinsicWidth);
+      final _ComboBoxBodyParentData childParentData = child.parentData! as _ComboBoxBodyParentData;
+      child = childParentData.nextSibling;
+    }
+
+    return math.max(width, _kMinimumWidth);
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    RenderBox? child = firstChild;
+    double width = 0;
+    while (child != null) {
+      if (child == firstChild) {
+        final _ComboBoxBodyParentData childParentData = child.parentData! as _ComboBoxBodyParentData;
+        child = childParentData.nextSibling;
+        continue;
+      }
+      final double maxIntrinsicWidth = child.getMaxIntrinsicWidth(height);
+      // Add the width of leading Icon.
+      if (child == lastChild) {
+        width += maxIntrinsicWidth;
+      }
+      // Add the width of trailing Icon.
+      if (child == childBefore(lastChild!)) {
+        width += maxIntrinsicWidth;
+      }
+      width = math.max(width, maxIntrinsicWidth);
+      final _ComboBoxBodyParentData childParentData = child.parentData! as _ComboBoxBodyParentData;
+      child = childParentData.nextSibling;
+    }
+
+    return math.max(width, _kMinimumWidth);
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double height) {
+    final RenderBox? child = firstChild;
+    double width = 0;
+    if (child != null) {
+      width = math.max(width, child.getMinIntrinsicHeight(height));
+    }
+    return width;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double height) {
+    final RenderBox? child = firstChild;
+    double width = 0;
+    if (child != null) {
+      width = math.max(width, child.getMaxIntrinsicHeight(height));
+    }
+    return width;
+  }
+
+  @override
+  bool hitTestChildren(BoxHitTestResult result, { required Offset position }) {
+    final RenderBox? child = firstChild;
+    if (child != null) {
+      final _ComboBoxBodyParentData childParentData = child.parentData! as _ComboBoxBodyParentData;
+      final bool isHit = result.addWithPaintOffset(
+        offset: childParentData.offset,
+        position: position,
+        hitTest: (BoxHitTestResult result, Offset transformed) {
+          assert(transformed == position - childParentData.offset);
+          return child.hitTest(result, position: transformed);
+        },
+      );
+      if (isHit) {
+        return true;
+      }
+    }
+    return false;
+  }
+}
+
+class _ComboBoxDefaultsM3 extends ComboBoxThemeData {
```

Comment:
We don't have tokens for these because we don't have a spec, right? A comment that these are hand coded until we have tokens/spec would be good here.

Reply:
Yes, and added a comment. Thanks!